config/include.go

```diff
@@ -387,11 +387,20 @@ func (targetConfig *TerragruntConfig) DeepMerge(sourceConfig *TerragruntConfig,
 	// Skip has to be set specifically in each file that should be skipped
 	targetConfig.Skip = sourceConfig.Skip
 
-	// Handle list attributes by concatenatenation
+	// Copy only dependencies which doesn't exist in source
 	if sourceConfig.Dependencies != nil {
 		resultModuleDependencies := &ModuleDependencies{}
 		if targetConfig.Dependencies != nil {
-			resultModuleDependencies.Paths = targetConfig.Dependencies.Paths
+			// take in result dependencies only paths which aren't defined in source
+			// Fix for issue: https://github.com/gruntwork-io/terragrunt/issues/1900
+			targetPathMap := fetchDependencyPaths(targetConfig)
+			sourcePathMap := fetchDependencyPaths(sourceConfig)
+			for key, value := range targetPathMap {
+				_, found := sourcePathMap[key]
+				if !found {
+					resultModuleDependencies.Paths = append(resultModuleDependencies.Paths, value)
+				}
+			}
```

Comment:
Ah this logic works if there are no `dependencies` blocks, but since we still support that, this would break it.

You would need to somehow take into account paths that are in `targetConfig.Dependencies.Paths` which are not associated with any item in `TerragruntDependencies`.